src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
```suggestion
  const children = people.filter(person => (
  person.sex === 'm'
      && person.some(mother => mother.name === person.mother))
      : people.some(mother => mother.name === person.mother));
```
Duplicated code fragment